core/templates/dev/head/pages/collection_editor/settings_tab/CollectionPermissionsCardDirective.js

```diff
@@ -22,10 +22,24 @@ oppia.directive('collectionPermissionsCard', [function() {
     restrict: 'E',
     templateUrl: 'inline/collection_permissions_card_directive',
     controller: [
-      '$scope',
-      function($scope) {
-        $scope.ownerNames = GLOBALS.ownerNames;
-        $scope.isPrivate = GLOBALS.isPrivate;
+      '$scope', 'CollectionRightsObjectFactory',
+      'CollectionRightsBackendApiService',
+      function($scope, CollectionRightsObjectFactory,
+        CollectionRightsBackendApiService) {
+        $scope.collectionId = GLOBALS.collectionId;
+        $scope.collectionRightsObject = null;
+        $scope.hasRightsLoaded = false;
+        $scope.isPrivate =
+          CollectionRightsBackendApiService.isCollectionPrivate;
+        var refreshPermissionsCard = function() {
+          CollectionRightsBackendApiService.loadCollectionRights(
+            $scope.collectionId).then(function(data) {
+            $scope.collectionRightsObject =
```

Comment:
If we are binding to values in the ``collectionRightsObject``, then it should have a method for updating itself. Look at ``Collection.copyFromCollection()`` in ``CollectionObjectFactory``. The idea here is we can have Angular bindings on domain object properties that are immutable, then update an instance of a particular rights object by copying it from another (this keeps our reference and doesn't break existing Angular bindings). It technically violates immutability, but we're not dealing with a multi-threaded environment so that's not a big deal.

I recommend you also add tests for such a copy method, and make sure you're copying the owner names array without replacing its reference. Happy to answer any questions to have on this particular suggestion.